### PR TITLE
handle ECONNRESET, add 'Zzz' when sleeping, log the response to a file with 'trace' option

### PR DESCRIPTION
**Pull request by niraj8**
Handle ECONNRESET as a DNS fail

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -355,13 +355,13 @@ var Util = function(opts) {
...
               // These are errors we can address by retrying
-              var errs = ['ENOTFOUND', 'ETIMEDOUT', 'ESOCKETTIMEDOUT']
+              var errs = ['ENOTFOUND', 'ETIMEDOUT', 'ESOCKETTIMEDOUT', 'ECONNRESET']
               if (errs.includes(code) && retry) {
                 retries += 1
                 // POST is dangerous to retry whatever the circumstances
                 if (retries < retry && method !== 'post') {
                   shouldRetry = true
-                  if (code === 'ENOTFOUND') {
+                  if (code === 'ENOTFOUND' || code === 'ECONNRESET') {
```
Comment (bramakris) [inline index.js]: We'll have to handle this separately I think, as it doesn't fall under DNS failures or timeouts. Can you add a separate var to track this, and add it to the log as well? 
`socketResets` is a good name, I think

Comment (niraj8) [reply index.js]: Agreed.

File: index.js [hunk 2]
```diff
@@ -332,6 +334,7 @@ var Util = function(opts) {
             _.token()
             .then(token => API[meth + 'Collection'](name, token, obj))
             .then(res => {
+              if (opts.trace) logfile.write(JSON.stringify(res, null, 0)+'\n')
```
Comment (bramakris) [inline index.js]: A date in ISO format would help ping support with the correct time in case of errors

Comment (niraj8) [reply index.js]: adding it to the start to line.
eg. 
[2018-03-29T06:45:14.890Z]  res or err here

**Pull request merged**